### PR TITLE
feat: implement `noInteractiveElementToNoninteractiveRole` options

### PR DESCRIPTION
**Pull request by vasucp1207**
## Summary

Add options to rule,
```js
"options": {
       "elementAndRoles": [
           { "element": "span", "roles": ["button", "grid"] },
           { "element": "div", "roles": ["button"] },
        ]
}
```

## Test Plan

No test added

### CODE REVIEW

**Comment by ematipico** [comment]
Thank you @vasucp1207, for the contribution. Can you update the description by explaining what these options are for and what are the uses cases? Usually, we tend to add options **only if they are needed**. We believe it's best for users to suppress a rule instead of adding an option as "expectation".

**Comment by vasucp1207** [comment]
> Thank you @vasucp1207, for the contribution. Can you update the description by explaining what these options are for and what are the uses cases? Usually, we tend to add options **only if they are needed**. We believe it's best for users to suppress a rule instead of adding an option as "expectation".

Oh! I just added this to provide an extra functionality and I agree that providing options to this rule is not a good idea.

**Comment by nissy-dev** [comment]
I also think there is no need to add options to `noInteractiveElementToNoninteractiveRole` and the example mentioned in ESLint docs is very rare case.

By the way, I think it's good to add an option to `noNoninteractiveElementToInteractiveRole`. (**not noInteractiveElementToNoninteractiveRole**)  There are common use cases for this option. `<ul role="listbox">` and `<ul role="menu" />`, mentioned in the ESLint documentation are actually used to mark up accessible dropdowns. It makes sense that we can ignore a rule with an option.

In summary, when submitting a PR for an option, please include the specific use case in the PR body as well.
